src/main/kotlin/com/cjbooms/fabrikt/generators/client/SpringHttpInterfaceGenerator.kt

```diff
@@ -0,0 +1,273 @@
+            val acceptHeaders = headerParams.filter { header ->
+                header.typeInfo is KotlinTypeInfo.Enum && header.typeInfo.entries.size == 1
+            }.filter { header ->
+                header.name == ClientGeneratorUtils.ACCEPT_HEADER_NAME
+            }.map { header ->
+                header.typeInfo as KotlinTypeInfo.Enum
+                buildCodeBlock {
+                    add("%S", header.typeInfo.entries.first())
+                }
+            }.takeIf { it.isNotEmpty() }
+                ?: run {
+                    // Add default accept header
+                    val block = operation.getPrimaryContentMediaType()?.key?.let { mediaType ->
+                        buildCodeBlock {
+                            add("%S", mediaType)
+                        }
+                    }
+                    listOfNotNull(block)
+                }
+
+            if (acceptHeaders.isNotEmpty()) {
+                addMember("accept=%L", acceptHeaders)
+            }
+        }
+
+        private fun AnnotationSpec.Builder.addHeaders(
+            headerParams: List<RequestParameter>,
+        ): AnnotationSpec.Builder = apply {
+            val headerValues = headerParams.filter { header ->
+                header.typeInfo is KotlinTypeInfo.Enum && header.typeInfo.entries.size == 1
+            }.filterNot { header ->
+                header.name == ClientGeneratorUtils.ACCEPT_HEADER_NAME
+            }.map { header ->
+                header.typeInfo as KotlinTypeInfo.Enum
+                buildCodeBlock {
+                    add("%S=%S", header.originalName, header.typeInfo.entries.first())
+                }
+            }
+
+            if (headerValues.isNotEmpty()) {
+                addMember("headers=%L", headerValues)
+            }
+        }
+
+        private fun List<IncomingParameter>.getHeaderParameters(): List<RequestParameter> =
+            filterIsInstance<RequestParameter>()
+                .filter { it.parameterLocation is HeaderParam }
+
+        private fun List<IncomingParameter>.getPathParameters(): List<RequestParameter> =
+            filterIsInstance<RequestParameter>()
+                .filter { it.parameterLocation is PathParam }
+
+        private fun resourceWithDeduplicatedParamNames(
+            resource: String,
+            pathParameters: List<RequestParameter>,
+        ): String {
+            var resolvedResource = resource
+            pathParameters.forEach { parameter ->
+                if (parameter.originalName != parameter.name) {
+                    resolvedResource = resolvedResource.replace("{${parameter.originalName}}", "{${parameter.name}}")
+                }
+            }
+            return resolvedResource
+        }
+    }
+
+    override fun generateLibrary(options: Set<ClientCodeGenOptionType>): Collection<GeneratedFile> = setOf(
+        generatedAnnotationFile()
+    )
+
+    private fun generatedAnnotationFile(): SimpleFile {
```

Comment:
The `@GeneratedClient` annotation is not required for the feature itself, but SpringBoot often has use cases where classes are searched for using annotations, and this is what it is for

Reply:
Where is this annotation defined? I cannot see any class imports

Reply:
It is generated in the same location as the client's interface.
Since they are in the same package, there should be no need for an import statement.

Reply:
I'm not so sure about this, it feels wrong to be generating a random annotation just in case it is useful. Could developers instead define their own annotation and add it to the implementation classes?

Reply:
For example, like creating an option like --http-client-annotations <annotation FQCN> and annotating the generated clients?

 It seems useful not only for this client type, so I'll do it in another PR and revert it in this one.

Reply:
Opened #386 and revert `@GeneratedClient` in this PR